### PR TITLE
Thresholded Linear Unit

### PR DESCRIPTION
**Pull request by AakashKumarNain**
This activation unit is introduced in Filter Response Normalization

### CODE REVIEW

**Comment by Squadrick** [comment]
@AakashKumarNain Any reason to put this under `layers` as opposed to `activations`? Also, could you update the `README.md`?

File: tensorflow_addons/layers/tlu.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
Comment (Squadrick) [inline tensorflow_addons/layers/tlu.py]: Nit: 2019 -> 2020

File: tensorflow_addons/layers/tlu_test.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
Comment (Squadrick) [inline tensorflow_addons/layers/tlu_test.py]: Nit: 2019 -> 2020

File: tensorflow_addons/layers/tlu.py (hunk 1 above)
Comment (AakashKumarNain) [reply tensorflow_addons/layers/tlu.py]: LOL. Yeah I will change that.

**Comment by AakashKumarNain** [comment]
@Squadrick we have two learnable parameters for this activation, similar to `PReLU`. It makes more sense to build a layer for this and if we want to include it in the `activations` package, then the activation function should initalize an instance of this layer first and then call the activation function on the inputs. Lemme know what you think

**Comment by Squadrick** [comment]
@AakashKumarNain Yeah, you're right. I'd forgotten that `tf.keras` expects activations to be stateless. Let's not bother with creating anything under `activations`. There's a precedent with core `tf.keras` making PReLU a layer.

File: tensorflow_addons/layers/tlu.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            initializer=self.tau_initializer,
+            regularizer=self.tau_regularizer,
+            constraint=self.tau_constraint)
+        if self.affine:
+            self.alpha = self.add_weight(
+                shape=param_shape,
+                name='alpha',
+                initializer=self.alpha_initializer,
+                regularizer=self.alpha_regularizer,
+                constraint=self.alpha_constraint)
```
Comment (Squadrick) [inline tensorflow_addons/layers/tlu.py]: It would be better to set `aggregation` and `synchronization` for `add_weight()` as well. This will make it suitable for distributed training out of the box, instead of having to deal with that in the future. [Link](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer?version=nightly#add_weight) to docs.

`synchronization=AUTO`, and `aggregation=MEAN` should be ideal.

Comment (AakashKumarNain) [reply tensorflow_addons/layers/tlu.py]: Cool. Will do that.

**Comment by AakashKumarNain** [comment]
> @AakashKumarNain Also test for serialization.

Can you elaborate on this? Any example?

File: tensorflow_addons/layers/tlu.py (hunk 3 above)
Comment (AakashKumarNain) [reply tensorflow_addons/layers/tlu.py]: `aggregation=MEAN` Shouldn't we leave this to the user for custom training loops?

Comment (Squadrick) [reply tensorflow_addons/layers/tlu.py]: Nope, it's more related to how you expect the variable to behave. You would only take the mean of these variables, similar to how you would take the mean of the weights of the conv layer, etc. 

I can't think of a situation where you would set this to either `FIRST_REPLICA_ONLY` or `SUM`. But if you do, let me know, and we can reconsider setting this to `MEAN` by default.

**Comment by Squadrick** [comment]
> > @AakashKumarNain Also test for serialization.
> 
> Can you elaborate on this? Any example?

https://github.com/tensorflow/addons/blob/2bcf8c874b18b88f454a758e0f64b959c8543fda/tensorflow_addons/layers/wrappers_test.py#L99

**Comment by AakashKumarNain** [comment]
Will this suffice?

```python
def test_serialization(self):
    tlu = TLU(affine=True, alpha_initializer='ones', tau_initializer='ones')
    serialized_tlu = tf.keras.layers.serialize(tlu)
    new_layer = tf.keras.layers.deserialize(serialized_tlu, custom_objects={'TLU':TLU})
    self.assertEqual(tlu.get_config(), new_layer.get_config())
```

File: tensorflow_addons/layers/tlu.py (hunk 3 above)
Comment (AakashKumarNain) [reply tensorflow_addons/layers/tlu.py]: Hmm agreed

**Comment by Squadrick** [comment]
@AakashKumarNain Yeah, that works. It'll work without the `custom_objects` as well.

**Comment by AakashKumarNain** [comment]
@seanpmorgan @Squadrick I think we can merge it now

**Comment by AakashKumarNain** [comment]
Thanks

**Comment by AakashKumarNain** [comment]
I forgot to update the README. Will do it later.
